free5gc-operator/api/v1alpha1/smfdeployment_types.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+type InterfaceConfig struct {
+	Name       string   `json:"name"`
+	IPs        []string `json:"ips"`
+	GatewayIPs []string `json:"gatewayIPs"`
+}
+
+type SMFCapacity struct {
+	MaxPduSessions int `json:"maxpdusessions"`
+	// +optional
+	MaxConnectedUpfs int `json:"maxconnectedupfs"`
+}
+
+type SbiEndpoint struct {
+    IPAddr   string `json:"ipaddr"`
+    Port     int    `json:"port"`
+    // +optional
+    NrfRegIPAddr string `json:"nrfregipaddr"`
+}
+
+type SbiSpec struct {
+	Protocol  string        `json:"protocol"` // should defaults to http
+	Endpoints []SbiEndpoint `json:"endpoints"`
+}
+
+
+
+// SMFDeploymentSpec defines the desired state of SMFDeployment
+type SMFDeploymentSpec struct {
```

Comment:
I am guessing this will all change once https://github.com/nephio-project/api/pull/11 merges?

Reply:
Yes this PR needs changes in line with https://github.com/nephio-project/api/pull/11